### ISSUE TITLE
Model.fit(..., **kwargs) unexpected behaviour

### ISSUE THREAD

**Issue by JacoGasp** [keras-team/keras#9805]
Hello,

I have a big dictionary with several parameters for my model implementation, including parameters for for the Model.fit() method but not only.
 Since the Model.fit() signature has the `**kwargs` key, I would expect that passing a dict with more parameters than the ones request by Model.fit(), it should simply infer which parameters take.

Here a dummy example:
```python
class  Model(Container):
   ...
   def fit(a, b, **kwargs)

my_parameters = {
   "a": 1,
   "b": 2,
   "c": 3
}

model = Model(...)
model.fit(**my_parameters)
```
This should simply works. Instead, at line 1619 of training.py, in the Model.fit() implementation there is
```python
if kwargs:
raise TypeError('Unrecognized keyword arguments: ' + str(kwargs))
```
that throws the exception
`TypeError('Unrecognized keyword arguments: 'c')`

Commenting those two line, everything works perfectly.
The question is: is there any reason for that if clause? Why is it present and can it be removed?

**Comment by gabrieldemarmiesse** [comment]
This close is for other arguments listed in the documentation but not in the function signature. What would you like to put in this dictionary?

**Comment by JacoGasp** [comment]
Hi @gabrieldemarmiesse,
 
Let's say that I have a class in which there are several methods, each one with its own parameters, including the fit method, other keras methods and more.
I would like to initialize the class with just one big dict containing _all_ the parameters for _all_ the methods, i.e. the learning rate, number of epochs etc, but also parameters used in some custom methods (size of the dataset, number of hidden nodes and so on).
In this way I can tune all in once, just modifying the dictionary. 
Thanks to the python paradigm of the `**kargs` keyword, it would be nice if I can just pass the dict to every method without specifying the parameters one by one for each single method of the class, but letting the method decide which parameter it needs.
Since the fit method already has the magic keyword in the signature, I don't understand why there is this restriction.

It is only for brevity of the code, nothing more.

**Comment by gabrieldemarmiesse** [comment]
I see what you mean. Indeed, it would be helpful in your case. Keras is very careful about user inputs. Always checking that there is no unexpected behaviour (like checking you don't pass arguments that are going to be ignored). So your only option, sadly, is splitting the dictionary before putting it in the fit function.

**Comment by JacoGasp** [comment]
Ok, thanks.
